benchmark/lskv_cluster.py

```diff
@@ -0,0 +1,677 @@
+        Make the config of a joining node.
+        """
+        enclave_file = "/app/liblskv.virtual.so"
+        enclave_type = "Virtual"
+        if self.enclave == "sgx":
+            enclave_file = "/app/liblskv.enclave.so.signed"
+            enclave_type = "Release"
+        app_protocol = "HTTP1" if self.http_version == 1 else "HTTP2"
+
+        base_client_port = 8000
+        return {
+            "enclave": {"file": enclave_file, "type": enclave_type},
+            "network": {
+                "node_to_node_interface": {
+                    "bind_address": f"{self.ip_address}:{self.peer_port}"
+                },
+                "rpc_interfaces": {
+                    "main_interface": {
+                        "bind_address": f"0.0.0.0:{self.client_port}",
+                        "app_protocol": app_protocol,
+                    }
+                },
+            },
+            "node_certificate": {"subject_alt_names": ["iPAddress:127.0.0.1"]},
+            "command": {
+                "type": "Join",
+                "service_certificate_file": "/app/common/service_cert.pem",
+                "join": {
+                    "target_rpc_address": f"{self.first_ip}:{base_client_port}",
+                },
+            },
+            "worker_threads": self.worker_threads,
+            "snapshots": {
+                "tx_count": self.snapshot_tx_interval,
+            },
+            "ledger_signatures": {
+                "tx_count": self.sig_tx_interval,
+                "delay": f"{self.sig_ms_interval}ms",
+            },
+            "ledger": {"chunk_size": self.ledger_chunk_bytes},
+        }
+
+
+# pylint: disable=too-many-instance-attributes
+class Operator:
+    """
+    Operator for a network of nodes.
+    """
+
+    # pylint: disable=too-many-arguments
+    def __init__(
+        self,
+        workspace: str,
+        image: str,
+        enclave: str,
+        http_version: int,
+        worker_threads: int,
+        sig_tx_interval: int,
+        sig_ms_interval: int,
+        ledger_chunk_bytes: str,
+        snapshot_tx_interval: int,
+    ):
+        self.workspace = workspace
+        self.name = "lskv"
+        self.nodes: List[Node] = []
+        self.image = image
+        self.enclave = enclave
+        self.http_version = http_version
+        self.subnet_prefix = "172.20.5"
+        self.worker_threads = worker_threads
+        self.sig_tx_interval = sig_tx_interval
+        self.sig_ms_interval = sig_ms_interval
+        self.ledger_chunk_bytes = ledger_chunk_bytes
+        self.snapshot_tx_interval = snapshot_tx_interval
+        self.create_network()
+
+    def create_network(self):
+        """
+        Create a Docker network for the nodes.
+        """
+        run(
+            [
+                "docker",
+                "network",
+                "create",
+                "--subnet",
+                f"{self.subnet_prefix}.0/16",
+                "lskv",
+            ]
+        )
```

Comment:
There's [a nice Python library for docker](https://docker-py.readthedocs.io/en/stable/) that CCF uses [here](https://github.com/microsoft/CCF/blob/main/tests/infra/remote_shim.py#L62). I don't think it's worth changing your code to use it, just an FYI.